src/test/java/uk/gov/homeoffice/digital/sas/balancecalculator/kafka/consumer/BalanceCalculatorConsumerServiceTest.java

```diff
@@ -0,0 +1,107 @@
+import static uk.gov.homeoffice.digital.sas.balancecalculator.constants.TestConstants.MESSAGE_INVALID_RESOURCE;
+import static uk.gov.homeoffice.digital.sas.balancecalculator.constants.TestConstants.MESSAGE_INVALID_VERSION;
+import static uk.gov.homeoffice.digital.sas.balancecalculator.constants.TestConstants.MESSAGE_VALID_RESOURCE;
+import static uk.gov.homeoffice.digital.sas.balancecalculator.constants.TestConstants.MESSAGE_VALID_VERSION;
+
+@SpringBootTest
+@ExtendWith({OutputCaptureExtension.class})
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
+class BalanceCalculatorConsumerServiceTest {
+
+  @Autowired
+  BalanceCalculatorConsumerService balanceCalculatorConsumerService;
+
+  TimeEntry expectedTimeEntry = TestUtils.createTimeEntry();
+
+  @Test
+  void onMessage_deserializeKafkaMessageAndLogSuccess_when_validMessageIsReceived
+      (CapturedOutput capturedOutput) {
+
+    // given
+    String message = TestUtils.createKafkaMessage(MESSAGE_VALID_RESOURCE, MESSAGE_VALID_VERSION);
+
+    // when
+    balanceCalculatorConsumerService.onMessage(message);
+    // then
+    assertThat(balanceCalculatorConsumerService.getKafkaEventMessage().getSchema()).isEqualTo(MESSAGE_EXPECTED_SCHEMA);
+    assertThat(balanceCalculatorConsumerService.getKafkaEventMessage().getAction()).isEqualTo(KafkaAction.CREATE);
+    assertMessageIsDeserializedAsExpected();
+    assertThat(capturedOutput.getOut()).contains(String.format(KAFKA_SUCCESSFUL_DESERIALIZATION,
+        expectedTimeEntry.getId()));
+  }
+
+  @Test
+  void onMessage_notDeserializeKafkaMessageAndLogFailure_when_inValidMessageIsReceived
+      (CapturedOutput capturedOutput) {
+    //given
+    String message = TestUtils.createKafkaMessage(MESSAGE_VALID_RESOURCE, MESSAGE_INVALID_VERSION);
+    // when
+    balanceCalculatorConsumerService.onMessage(message);
+    // then
+    assertThat(balanceCalculatorConsumerService.getKafkaEventMessage()).isNull();
+    assertThat(capturedOutput.getOut()).contains(String.format(KAFKA_UNSUCCESSFUL_DESERIALIZATION,
+        message));
+  }
+
+  @Test
+  void onMessage_notDeserializeKafkaMessage_when_schemaNotTimeEntry(CapturedOutput capturedOutput){
+    String message = TestUtils.createKafkaMessage(MESSAGE_INVALID_RESOURCE,
+        MESSAGE_VALID_VERSION);
+
+    balanceCalculatorConsumerService.onMessage(message);
+
+    assertThat(balanceCalculatorConsumerService.getKafkaEventMessage()).isNull();
+    assertThat(capturedOutput.getOut()).contains(String.format(KAFKA_UNSUCCESSFUL_DESERIALIZATION,
+        message));
+  }
+
+  private void assertMessageIsDeserializedAsExpected() {
+
+    KafkaEventMessage<TimeEntry> expectedKafkaEventMessage =
+        TestUtils.generateExpectedKafkaEventMessage(MESSAGE_VALID_VERSION,
+        expectedTimeEntry,
+        KafkaAction.CREATE);
+
+    assertThat(balanceCalculatorConsumerService.getKafkaEventMessage().getSchema()).isEqualTo(MESSAGE_VALID_RESOURCE + ", " + MESSAGE_VALID_VERSION);
```

Comment:
Don't think this should be called here, the message should be passed as an argument but calling the method twice is also not correct

Reply:
Due to the refactor we no longer have direct access to the kafkaEventMessage object. These methods have been removed.